grails-webhooks/grails-app/services/webhooks/WebhookService.groovy

```diff
@@ -159,6 +159,7 @@ class WebhookService {
         hook.uuid = hookData.uuid ?: hook.uuid
         hook.name = hookData.name ?: hook.name
         hook.project = hookData.project ?: hook.project
+        if(hookData.secret != null) hook.secret = hookData.secret
```

Comment:
should the secret be stored as a hashed value if it is not a key storage path?

Reply:
If we hash the secret we can't display the secret on edit, so the user can't go back and check the secret value. Definitely more secure, but not sure if the UI would make sense.

Reply:
I vote hash it and they can change it.

Reply:
We'll probably need to give the user a warning that when they put a value in the Secret/Authorization String field, if it doesn't refer to a key store value, it will be hashed and the original value will no longer be visible. Thoughts?

Reply:
So if it's a key-storage entry it would still be possible to see the string.

If it's not a key-storage entry it would be obscured with ***** right?

How will it know it's a key-storage string?  Should we could have a separate box for selecting from KS and then the straight string box would be **** even while they are typing.  That would imply they won't be able to see it in the future (like a password) and I can cover off the field being obscured in docs.

Reply:
I think we should not overload the input field with both key storage path and "secret" value.  maybe it's simpler to do one or the other.  Currently we don't have a simple UI for that kind of behavior.  I notice @sjrd218 that access to key storage depends on the auth context defined for the webhook request - will that auth context have correct read access to key storage?

Reply:
Then let's just use Key Storage.